merkle/tree_hasher.go

```diff
@@ -52,28 +52,24 @@ func NewRFC6962TreeHasher() TreeHasher {
 
 // HashEmpty returns the hash of an empty element for the tree
 func (t TreeHasher) HashEmpty() []byte {
-	return t.Digest([]byte{})
+	return t.New().Sum(nil)
 }
 
 // HashLeaf returns the Merkle tree leaf hash of the data passed in through leaf.
 // The data in leaf is prefixed by the LeafHashPrefix.
 func (t TreeHasher) HashLeaf(leaf []byte) []byte {
```

Comment:
Using the hasher interface is more efficient because `h.Write` can absorb inputs without first needing to copy all the inputs into a separate, contiguous input buffer.